### PR TITLE
Feature/cb 52, Feature/cb 63

### PR DESCRIPTION
**Pull request by DenilFuirie**

### CODE REVIEW

File: src/containers/Routes/index.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { CustomDevelopment } from 'pages/CustomDevelopment';
 import TokenContract from 'pages/TokenContract/TokenContract';
 import WeddingContract from 'pages/WeddingContract/WeddingContract';
 import TokenContractPreview from 'pages/TokenContractPreview/TokenContractPreview';
+import WeddingContractPreview from '../../pages/WeddingContractPreview/WeddingContractPreview';
```
Comment (ValnetineKul) [inline src/containers/Routes/index.tsx:12]: The aliases has been configured ../../pages === pages, change and always use just folder name in src as start of your route

File: src/pages/WeddingContract/WeddingContract.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { makeStyles } from '@material-ui/core';
+import { createStyles, Theme } from '@material-ui/core/styles';
+import {
+  COLOR_BLACK_1, COLOR_BLACK_3, COLOR_GREY_1, COLOR_GREY_2, COLOR_GREY_5,
+} from 'theme/colors';
+import { flexHelper } from '../../utils';
```
[the remark this answers is not in the document]
Comment (ValnetineKul) [reply src/pages/WeddingContract/WeddingContract.styles.tsx:6]: ?

[the remark this answers is not in the document]
Comment (ValnetineKul) [reply src/pages/WeddingContract/WeddingContract.styles.tsx:6]: import { flexHelper } from 'utils';

File: src/pages/WeddingContract/WeddingContract.tsx [hunk 3]
```diff
@@ -28,22 +29,26 @@ import { useStyles } from './WeddingContract.styles';
...
-  const [partnerTwoSliderValue, setPartnerTwoSliderValue] = useState(weddingContract.partnerTwoSliderValue);
+  const [
+    partnerOneSliderValue, setPartnerOneSliderValue,
+  ] = useState(weddingContract.partnerOneSliderValue);
+  const [
+    partnerTwoSliderValue, setPartnerTwoSliderValue,
+  ] = useState(weddingContract.partnerTwoSliderValue);
 
-  const onFirstSliderHandler = (_, value: number): void => {
+  const onFirstSliderHandler = useMemo(() => (_, value: number) => {
```
Comment (ValnetineKul) [inline src/pages/WeddingContract/WeddingContract.tsx:47]: useCallback for functions, useMemo for objects

File: src/pages/WeddingContract/WeddingContract.tsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+  const [partnerOneSliderValue, setPartnerOneSliderValue] = useState(weddingContract.partnerOneSliderValue);
+  const [partnerTwoSliderValue, setPartnerTwoSliderValue] = useState(weddingContract.partnerTwoSliderValue);
+
+  const onFirstSliderHandler = (_, value: number): void => {
+    setPartnerOneSliderValue(value);
+    setPartnerTwoSliderValue(100 - value);
+  };
+  
+  const onSecondSliderHandler = (_, value: number): void => {
```
[the remark this answers is not in the document]
Comment (ValnetineKul) [reply src/pages/WeddingContract/WeddingContract.tsx:52]: didnt wrap

File: src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { makeStyles } from '@material-ui/core';
+import { createStyles, Theme } from '@material-ui/core/styles';
+import { COLOR_ACID_GREEN, COLOR_BLACK_3, COLOR_GREY_2 } from 'theme/colors';
+import { flexHelper } from 'utils';
+
+export const useStyles = makeStyles((theme: Theme) => createStyles({
+  tokenContractInfoBlock: {
+    padding: `${theme.spacing(4)}px 0`,
```
Comment (ValnetineKul) [inline src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx:8]: this style will not work, add px to 0`${theme.spacing(4)}px 0px
`

File: src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        color: COLOR_ACID_GREEN,
+      },
+    },
+  },
+  approvalInfo: {
+    padding: 0,
+    width: '100%',
+    paddingTop: theme.spacing(4),
+    ...flexHelper('space-between'),
+    '& > div': {
```
Comment (ValnetineKul) [inline src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx:57]: give div a class and use class

File: src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import { flexHelper } from 'utils';
+
+export const useStyles = makeStyles((theme: Theme) => createStyles({
+  tokenContractInfoBlock: {
+    padding: `${theme.spacing(4)}px 0`,
+    '&:nth-child(2)': {
+      borderBottom: `1px solid ${theme.palette.type === 'dark' ? COLOR_BLACK_3 : COLOR_GREY_2}`,
+    },
+  },
+  previewValueBlock: {
```
Comment (ValnetineKul) [inline src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx:13]: this styles do nothing, delete

File: src/pages/WeddingContractPreview/WeddingContractPreview.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+const WeddingContractPreview = () => {
+  const {
+    weddingContract,
+  } = useShallowSelector<State, WeddingFormsState>(weddingFormsSelector.getWeddingForms);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const handleEdit = useCallback(() => {
+    navigate(routes['token-contract'].root);
```
Comment (ValnetineKul) [inline src/pages/WeddingContractPreview/WeddingContractPreview.tsx:26]: naviage to your contract

File: src/pages/WeddingContractPreview/WeddingContractPreview.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+  }, []);
+
+  const handleDelete = useCallback(() => {
+    dispatch(deleteTokenContractForm());
```
Comment (ValnetineKul) [inline src/pages/WeddingContractPreview/WeddingContractPreview.tsx:30]: delete your contract form

File: src/pages/WeddingContractPreview/WeddingContractPreview.tsx [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                    >
+                      {weddingContract[name]}
+                    </Typography>
+                  </Copyable>
+                  <Box className={classes.subInfo}>
+                    <Typography>{weddingContract[partnerSliderValueKey]}%</Typography>
+                    <Typography>{weddingContract[partnerEmailKey]}</Typography>
+                  </Box>
+                  {bottomInfo && (
+                  <Grid item xl={6} lg={8} md={8} xs={12} sm={12} className={classes.approvalInfo}>
```
Comment (ValnetineKul) [inline src/pages/WeddingContractPreview/WeddingContractPreview.tsx:87]: Please use breakpoint from smallest to biggest like xs, sm, md, lg, xl

File: src/store/weddingForms/reducer.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+export const weddingFormReducer = createSlice({
+  name: 'weddingForms',
+  initialState,
+  reducers: {
+    setWeddingContractForm: (state, action: PayloadAction<WeddingContract>) => ({
+      ...state,
+      weddingContract: action.payload,
+    }),
+    setWeddingFirstSliderValueForm: (state, action) => ({
```
Comment (ValnetineKul) [inline src/store/weddingForms/reducer.ts]: why do you need a separate action for 1 slider?

**Review by ValnetineKul — CHANGES_REQUESTED** [review]

File: src/pages/WeddingContractPreview/WeddingContractPreview.tsx [hunk 12]
```diff
@@ -8,12 +8,12 @@ import { Copyable } from 'components/Copyable';
 import { routes } from 'appConstants';
 import { useNavigate } from 'react-router-dom';
 import { useDispatch } from 'react-redux';
-import { deleteTokenContractForm } from 'store/contractForms/reducer';
 import { useStyles } from './WeddingContractPreview.styles';
 import {
   staticWeddingContractPreviewHelpers,
 } from './WeddingContractPreview.helpers';
 import contractFormsSelector from '../../store/contractForms/selectors';
+import { deleteWeddingContractForm } from '../../store/contractForms/reducer';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/WeddingContractPreview/WeddingContractPreview.tsx:16]: fixed

File: src/pages/WeddingContract/WeddingContract.styles.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  container: {
+    ...flexHelper('space-between', 'center'),
+    flexWrap: 'nowrap',
+    [theme.breakpoints.down('sm')]: {
+      flexWrap: 'wrap',
+    },
+  },
+  slider: {
+    padding: '23px 19px',
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/WeddingContract/WeddingContract.styles.tsx:41]: fixed

File: src/pages/WeddingContractPreview/WeddingContractPreview.tsx [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+  }, []);
+
+  const handleDelete = useCallback(() => {
+    dispatch(deleteWeddingContractForm());
+    navigate(routes.root);
+  }, []);
+
+  const classes = useStyles();
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/WeddingContractPreview/WeddingContractPreview.tsx:36]: fixed

File: src/pages/WeddingContract/WeddingContract.tsx [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+import { State, IWeddingContract as WeddingContractType, ContractFormsState } from 'types';
+import { useShallowSelector } from 'hooks';
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import {
+  validationSchema,
+  weddingContractFormConfigEnd, weddingContractFormConfigStart,
+} from './WeddingContract.helpers';
+import { useStyles } from './WeddingContract.styles';
+import { routes } from '../../appConstants';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/WeddingContract/WeddingContract.tsx:28]: fixed

File: src/pages/WeddingContract/WeddingContract.styles.tsx [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+  },
+
+  helperText: {
+    marginTop: theme.spacing(4),
+  },
+  newCount: {
+    color: theme.palette.type === 'dark' ? COLOR_GREY_1 : COLOR_BLACK_1,
+  },
+  container: {
+    ...flexHelper('space-between', 'center'),
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/WeddingContract/WeddingContract.styles.tsx]: fixed

File: src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx (hunk 5 above)
Comment (DenilFuirie) [reply src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx:10]: fixed

File: src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx (hunk 6 above)
Comment (DenilFuirie) [reply src/pages/WeddingContractPreview/WeddingContractPreview.styles.tsx:59]: fixed

**Pull request merged**